igb_client/client.py

```diff
@@ -0,0 +1,189 @@
+import base64
+import copy
+from datetime import timedelta, datetime
+from typing import List, Dict, Optional, Union, Type, TypeVar
+
+import requests_cache
+
+from igb_client.dataclasses import JobBoard, ContractCredential, Credential, OfccpCredential, \
+    XMLSerializable
+from igb_client.encrypt import AESCypher
+from igb_client.parse import parse_igb_xml_payload
+
+CredentialInterface = TypeVar('CredentialInterface', bound=Credential)
+
+
+class IGBClientError(Exception):
+    pass
+
+
+class IGBClientBase:
+    _instance = None
+    _base_url = "{base_url}/{environment_id}/{view}"
+    _credentials_storage_key = None
+    _credentials_transport_key = None
+
+    def __init__(self, api_key: str, environment_id: str, credentials_storage_key: str,
+                 credentials_transport_key: str, base_url="https://api.ingoedebanen.nl/apipartner/hapi/v1",
```

Comment:
base_url could be defined as a constant outside the class:
BASE_URL  = https://...